components/prism-ini.js

```diff
@@ -1,11 +1,37 @@
 Prism.languages.ini= {
-	'comment': /^[ \t]*[;#].*$/m,
-	'selector': /^[ \t]*\[.*?\]/m,
-	'constant': /^[ \t]*[^\s=]+?(?=[ \t]*=)/m,
-	'attr-value': {
-		pattern: /=.*/,
+	'comment': {
+		pattern: /(^[ \f\t\v]*)[#;][^\n\r]*/m,
+		lookbehind: true
+	},
+	'header': {
+		pattern: /(^[ \f\t\v]*)\[[^\n\r\]]*\]?/m,
+		lookbehind: true,
 		inside: {
-			'punctuation': /^[=]/
+			'section-name': {
+				pattern: /(\[[ \f\t\v]*)[^ \f\n\r\t\v\]]+(?:[ \f\t\v]+[^ \f\n\r\t\v\]]+)*/,
+				lookbehind: true,
+				alias: 'selector'
+			},
+			rest: {
+				'punctuation': /\[|\]/
+			}
```

Comment:
`rest` will (in a sense) inline its grammar. You don't need `rest` in this case.

```suggestion
			'punctuation': /\[|\]/
```

Reply:
Right! Thanks!